daisy/resource_registry.go

```diff
@@ -284,3 +284,15 @@ func (dr *diskRegistry) regUseDeviceName(deviceName, project, zone, instance str
 	res.users = append(res.users, s)
 	return res, isAttached, nil
 }
+
+func (dr *diskRegistry) findDiskResourceByDeviceName(deviceName, instance string) (*Resource, DError) {
+	attachmentMap, ok := dr.attachments[deviceName]
+	if !ok {
+		return nil, Errf("missing registered disk attachment for device name '%v'", deviceName)
```

Comment:
Any reason to switch away from the original error format (the one that uses dr.typeName)?

Reply:
After changed to the new lookup method, the original error msg is not a correct description any more. The new msg described what is missed.